### DISCUSSION TITLE
iOS Push Notifications have stopped working

### DISCUSSION THREAD

**Discussion by knightian** [dani-garcia/vaultwarden#3967] [Q&A]
### Discussed in https://github.com/dani-garcia/vaultwarden/discussions/3945

<div type='discussions-op-text'>

<sup>Originally posted by **knightian** October  2, 2023</sup>
### Discussed in https://github.com/dani-garcia/vaultwarden/discussions/3943

<div type='discussions-op-text'>

<sup>Originally posted by **knightian** October  2, 2023</sup>
Latest version of vaultwarden and web vault

If I make a secure note in the webvault, it does not appear soon in the app on iOS.

I did a fresh generation of installation id and password for the config, and then I deleted the iOS app and reinstalled it so it would set up the push notifications in the app as per the wiki, but it does not work.

It used to work in previous versions.

There are no DNS issues at the moment because I am putting nameserver entry in resolv.conf to make trust-dns happy:

![image](https://github.com/dani-garcia/vaultwarden/assets/72685601/13fec400-72b1-4533-a002-39493f4a7091)

### Your environment (Generated via diagnostics page)
* Vaultwarden version: v1.29.2-bc26bfa5
* Web-vault version: v2023.8.2
* OS/Arch: linux/x86_64
* Running within Docker: false (Base: Not applicable)
* Environment settings overridden: false
* Uses a reverse proxy: true
* IP Header check: true (X-Real-IP)
* Internet access: false
* Internet access via a proxy: false
* DNS Check: true
* Browser/Server Time Check: true
* Server/NTP Time Check: n/a
* Domain Configuration Check: true
* HTTPS Check: true
* Database type: SQLite
* Database version: 3.41.2
* Clients used: 
* Reverse proxy and version: 
* Other relevant information: 

### Config (Generated via diagnostics page)
<details><summary>Show Running Config</summary>

**Environment settings which are overridden:** 

```json
{
  "_duo_akey": null,
  "_enable_duo": false,
  "_enable_email_2fa": true,
  "_enable_smtp": true,
  "_enable_yubico": true,
  "_icon_service_csp": "",
  "_icon_service_url": "",
  "_ip_header_enabled": true,
  "_smtp_img_src": "cid:",
  "admin_ratelimit_max_burst": 2,
  "admin_ratelimit_seconds": 6,
  "admin_session_lifetime": 20,
  "admin_token": "***",
  "allowed_iframe_ancestors": "",
  "attachments_folder": "data/attachments",
  "auth_request_purge_schedule": "30 * * * * *",
  "authenticator_disable_time_drift": false,
  "data_folder": "data",
  "database_conn_init": "",
  "database_max_conns": 256,
  "database_timeout": 30,
  "database_url": "***************",
  "db_connection_retries": 20,
  "disable_2fa_remember": false,
  "disable_admin_token": false,
  "disable_icon_download": false,
  "domain": "*****://********************",
  "domain_origin": "*****://********************",
  "domain_path": "",
  "domain_set": true,
  "duo_host": null,
  "duo_ikey": null,
  "duo_skey": null,
  "email_attempts_limit": 5,
  "email_expiration_time": 1800,
  "email_token_size": 6,
  "emergency_access_allowed": false,
  "emergency_notification_reminder_schedule": "0 3 * * * *",
  "emergency_request_timeout_schedule": "0 7 * * * *",
  "enable_db_wal": true,
  "event_cleanup_schedule": "0 10 0 * * *",
  "events_days_retain": null,
  "extended_logging": true,
  "helo_name": null,
  "hibp_api_key": null,
  "icon_blacklist_non_global_ips": true,
  "icon_blacklist_regex": null,
  "icon_cache_folder": "data/icon_cache",
  "icon_cache_negttl": 259200,
  "icon_cache_ttl": 2592000,
  "icon_download_timeout": 10,
  "icon_redirect_code": 302,
  "icon_service": "internal",
  "incomplete_2fa_schedule": "30 * * * * *",
  "incomplete_2fa_time_limit": 3,
  "invitation_expiration_hours": 120,
  "invitation_org_name": "Mine",
  "invitations_allowed": true,
  "ip_header": "X-Real-IP",
  "job_poll_interval_ms": 30000,
  "log_file": "/var/log/vaultwarden/vaultwarden.log",
  "log_level": "info",
  "log_timestamp_format": "%Y-%m-%d %H:%M:%S.%3f",
  "login_ratelimit_max_burst": 3,
  "login_ratelimit_seconds": 4,
  "org_attachment_limit": null,
  "org_creation_users": "**********************",
  "org_events_enabled": false,
  "org_groups_enabled": false,
  "password_hints_allowed": false,
  "password_iterations": 700442,
  "push_enabled": true,
  "push_installation_id": "***",
  "push_installation_key": "***",
  "push_relay_uri": "https://push.bitwarden.com",
  "reload_templates": false,
  "require_device_email": true,
  "rsa_key_filename": "data/rsa_key",
  "send_purge_schedule": "0 5 * * * *",
  "sendmail_command": null,
  "sends_allowed": true,
  "sends_folder": "data/sends",
  "show_password_hint": false,
  "signups_allowed": false,
  "signups_domains_whitelist": "***********,********************",
  "signups_verify": true,
  "signups_verify_resend_limit": 20,
  "signups_verify_resend_time": 300,
  "smtp_accept_invalid_certs": false,
  "smtp_accept_invalid_hostnames": false,
  "smtp_auth_mechanism": "\"Login\"",
  "smtp_debug": false,
  "smtp_embed_images": true,
  "smtp_explicit_tls": null,
  "smtp_from": "*****************************",
  "smtp_from_name": "Mine",
  "smtp_host": "******************",
  "smtp_password": "***",
  "smtp_port": 587,
  "smtp_security": "starttls",
  "smtp_ssl": null,
  "smtp_timeout": 15,
  "smtp_username": "***********************************",
  "templates_folder": "data/templates",
  "tmp_folder": "data/tmp",
  "trash_auto_delete_days": 30,
  "trash_purge_schedule": "0 5 0 * * *",
  "use_sendmail": false,
  "use_syslog": false,
  "user_attachment_limit": 102400,
  "web_vault_enabled": true,
  "web_vault_folder": "web-vault/",
  "websocket_address": "0.0.0.0",
  "websocket_enabled": false,
  "websocket_port": 3012,
  "yubico_client_id": null,
  "yubico_secret_key": null,
  "yubico_server": null
}
```
</details>
</div></div>

**Comment by knightian** [comment]
Now multiple reports of this issue see: https://github.com/dani-garcia/vaultwarden/discussions/3945

**Reply by BlackDex** [reply]
Not sure what you want from us after reading the mentioned posts.
We can't really debug this. And of there was an error you would see it in the logs.

**Reply by bokkabonga** [reply]
As the devs stated there is not much to be done from this side, since the push service doesn´t return an error.

Also I don´t see whats the use of opening the third discussion by now and always just quoting the last one...

**Reply by knightian** [reply]
We don't even know if it is sending to the push server at all. Nobody has looked, if it stamped a debug log entry on attempt we would know for sure yes or no, but as it does not, we don't know.

I guess the push code was made by someone who is gone now and current maintainers do not understand the push code so have no appetite to investigate this issue, that's my take on it.

I will set up MiTM and see what I can find, but even if I find something it seems maintainers will have no way to fix it as the issue just keeps getting closed regardless even though it is a known and accepted issue.

**Reply by BlackDex** [reply]
Well, if that is your take on it, just don't use it.
And, it seems you do not read my comments, else your remark on not being sent wouldn't have been made. But hey, it's Open Source, you can lookup who made the code and who are the current maintainers and determine if they have knowledge or not.

For me, these kind of remarks do not make me look into this issue any quicker, maybe even the opposite.
It feels for me that you take my time for granted, which i find a bit disrespectful.

**Reply by knightian** [reply]
It's nothing about time. I am not saying you have to fix it. But you keep closing the issue so nobody will even try. How will any other rust devs that come along know the issue exists when you keep closing it? To me it seems like you don't know how to solve the issue so instead of keeping it as an open issue you are just closing it, and that's poor form IMO. Just because you can't solve the issue doesn't mean you should close it.

**Reply by BlackDex** [reply]
Assumptions. But ok.

My reason is that what i have tested so far, and also others, and other users who use ios have no issues. Which to me indicates it's probably something on the Bitwarden side, again, you tried all suggestions already, no errors are reported, so again what do you expect from us dev's without errors? I can keep saying that, if it's at the Bitwarden servers it's out of our range to debug.

**Reply by knightian** [reply]
The only thing I wanted was a one or two lines in the code when debug logging was set to on so it would print that it was sending push, so I could avoid setting up MITM trusting ca cert etc. And perhaps a dump of the https <-> in the trace level logs as well, it is why debug and trace logs exist, I get what you say about too much logging but that is why there are log levels: trace, debug, warn, info etc. so we can select the verbosity exactly for that reason. Of course it will not log it when it is not set to debug

**Reply by BlackDex** [reply]
I understand that to. But patients seem to be missing here as if we don't have anything else to do and the whole world is burning down because of this feature not working. Chill.

**Reply by knightian** [reply]
I have no problem with time or anything like that. It is only a minor annoyance, everything still works fine, I just pull down to sync no big deal.

My only issue here is that the github issue keeps getting marked as closed. I do not expect you to solve it or anything like that, it would be nice for the debug log entries but if that doesn't happen it doesn't happen. My concern is that I think the github issue should remain open until it is no longer an issue. Even if nobody is working on it, at least others can a) see its a known issue and b) provide their input on the issue i.e. what they have done to debug it etc.

I just don't see any reason to keep closing the issue especially now I am not the only one having it. When I was the only one I said ok, and I just ignored, but now others are also saying they are having it, I think definitely it should be raised as an issue.

**Reply by BlackDex** [reply]
If you would have checked the code there is enough debug in there.
https://github.com/dani-garcia/vaultwarden/blob/main/src/api/push.rs

It doesn't seem anything got triggered.
Also, you posted a screenshot of your mitm test, which indicates the POST didn't send any data, which is impossible i think looking at the code quickly.

**Reply by knightian** [reply]
Yes I removed my pic because I realised it was the response that had no content not the request.

Also looking at that code:
![image](https://github.com/dani-garcia/vaultwarden/assets/72685601/8dde2efc-9d11-419c-9ccd-9dde835177c9)

It seems that for push device registration there is no error logging or debug or anything, it looks like it could try to register a device for push and fail without error. But admittedly I am not a rust dev so I could well be wrong.

**Reply by BlackDex** [reply]
> Yes I removed my pic because I realised it was the response that had no content not the request.

Ok, do you now get my point? The request is sent, and we do not receive any response. So, what do you expect we would have returned? You kept insisting that you need more debug, but there is no more to give.

> It seems that for push device registration there is no error logging or debug or anything, it looks like it could try to register a device for push and fail without error. But admittedly I am not a rust dev so I could well be wrong.

If you were a Rust dev, you would know that it does produce an error if there was any other status then `200` returned. The `error_for_status()?` takes care of that, and if that produced an error, it would show in the logs.

So, i now do hope that you can see that it is almost (if not totally) impossible for us to debug this, since all data is send and all is out of our reach to trace or check anything.

**Comment by stefan0xC** [comment]
@BlackDex do you know why it says `Internet access: false` in the generated support string but the picture displays `Ok`? Is this not the same value `has_http_access`? (It might not be related to the issue at hand but I find it weird.)

**Reply by BlackDex** [reply]
@stefan0xC , good point, it shouldn't and can't even i think.
That value is used at all places, so, that is is strange.

What could be of course is that the original message had an updated images after some changes, but not an updated diagnostics string ;).
